pkgs/applications/office/gnucash/default.nix

```diff
@@ -1,106 +1,210 @@
+      test-guid \
+      test-import-account-matcher \
+      test-import-backend \
+      test-import-map \
+      test-import-parse \
+      test-import-pending-matches \
+      test-incompatdep \
+      test-job \
+      test-kvp-frames \
+      test-kvp-value \
+      test-link-module-tax-us \
+      test-link-ofx \
+      test-load-backend \
+      test-load-c \
+      test-load-engine \
+      test-load-example-account \
+      test-load-xml2 \
+      test-lots \
+      test-modsysver \
+      test-numeric \
+      test-object \
+      test-print-parse-amount \
+      test-qof \
+      test-qofquerycore \
+      test-qofsession \
+      test-query \
+      test-querynew \
+      test-recurrence \
+      test-resolve-file-path \
+      test-scm-query \
+      test-scm-query-string \
+      test-split-register-copy-ops \
+      test-split-vs-account \
+      test-sqlbe \
+      test-string-converters \
+      test-sx \
+      test-tokenizer \
+      test-transaction-reversal \
+      test-transaction-voiding \
+      test-userdata-dir \
+      test-userdata-dir-invalid-home \
+      test-vendor \
+      test-xml-account \
+      test-xml-commodity \
+      test-xml-pricedb \
+      test-xml-transaction \
+      test-xml2-is-file
+
+      export LD_LIBRARY_PATH="$PWD/lib:$PWD/lib/gnucash:$PWD/lib/gnucash/test:$PWD/lib/gnucash/test/future"
   '';
-  doCheck = true;
 
-  meta = {
-    description = "Personal and small-business financial-accounting application";
+  postInstall = ''
+    rm $out/bin/gnc-fq-update     # remove GnuCash update utility
+    rm $out/bin/gnucash-valgrind  # remove Valgrind
+  '';
 
-    longDescription = ''
-      GnuCash is personal and small-business financial-accounting software,
-      freely licensed under the GNU GPL and available for GNU/Linux, BSD,
-      Solaris, macOS and Microsoft Windows.
-
-      Designed to be easy to use, yet powerful and flexible, GnuCash allows
-      you to track bank accounts, stocks, income and expenses.  As quick and
-      intuitive to use as a checkbook register, it is based on professional
-      accounting principles to ensure balanced books and accurate reports.
-    '';
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --set GNC_DBD_DIR ${libdbiDrivers}/lib/dbd                                      # specify where db drivers are
+      --set GSETTINGS_SCHEMA_DIR ${glib.makeSchemaPath "$out" "${pname}-${version}"}  # specify where nix puts the gnome settings schemas
+    )
+  '';
+
+  /*
+  wrapGAppsHook will naively wrap the cli utils which need Perl wrapping, so turning off wrapGAppsHook.
+  Keeping it as an input allows lots of useful magic, though.
+  */
+  dontWrapGApps = true;
 
-    license = lib.licenses.gpl2Plus;
+  /*
+  gnc-fq-* are cli utils that come with GnuCash to help with financial quote sources.
+  They rely on Perl libraries, hence the Perl wrapping ceremony.
+  gnucash itself is being wrapped using all of the args constructed for wrapGAppsHook.
+  */
```

Comment:
```suggestion
  # gnc-fq-* are cli utils written in Perl hence the extra wrapping
  # gnucash itself is being wrapped using all of the args constructed for wrapGAppsHook.
```

Reply:
fixed in 072900f31f1598aca35feb66e4c553bbe69a9509